### PR TITLE
Fix message filter being put in wrong position by optimizer

### PR DESCRIPTION
**Pull request by vlad-ivanov-name**

### CODE REVIEW

File: josh-core/src/filter/opt.rs [hunk 1]
```diff
@@ -315,6 +315,8 @@ fn common_post(filters: &Vec<Filter>) -> Option<(Filter, Vec<Filter>)> {
             common_post.map(|c| (c, rest))
         } else if let Op::Prefix(_) = to_op(c) {
             common_post.map(|c| (c, rest))
+        } else if let Op::Message(..) = to_op(c) {
```
Comment (christian-schilling) [inline josh-core/src/filter/opt.rs:318]: I wonder why this was limited to prefix in the first place. Maybe we should look into allowing factorizing all filters here instead of adding more special cases 🤔 
Thoughts?

Comment (vlad-ivanov-name) [reply josh-core/src/filter/opt.rs:318]: Would prefer to merge this as a hotfix and then create an issue. For all filters I think we'd need a good test strategy, maybe something like guided fuzzing to discover "unsound" edge cases

Comment (christian-schilling) [reply josh-core/src/filter/opt.rs:318]: Makes sense

**Review by christian-schilling — APPROVED** [review]
